third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2599,8 +2599,10 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
             }
             let mut success = true;
             for (i, arg_ty) in arg_types.iter().enumerate() {
-                let arg_ty = if cand.get_operation().allows_ref_param_for_value() {
-                    // Drop reference type if there is any.
+                let arg_ty = if cand.get_operation().allows_ref_param_for_value()
+                    && self.mode != ExpTranslationMode::Impl
+                {
+                    // Drop reference type if there is necessary.
```

Comment:
I feel this comment is fairly vague. Please document why it is necessary to drop reference type in this situation.

Reply:
done